### ISSUE TITLE
reference implementation?

### ISSUE THREAD

**Issue by dalehenrich** [CampSmalltalk/Cypress#4]
If you take a look at the work I've done on [Cypress for Amber](https://github.com/CampSmalltalk/amber-cypress), you'll see that I've got a fairly vanilla port of the [Monticello package/snapshot/patch code](https://github.com/CampSmalltalk/amber-cypress/blob/master/st/Cypress-Definitions.st). This code is much more portable that either [STIG](https://github.com/CampSmalltalk/STIG) or [FileTree](https://github.com/dalehenrich/filetree) so it would make a good candidate as a reference implementation that can be used as a starting point for doing a dialect port.

Like [FileTree](https://github.com/dalehenrich/filetree), there's not a lot of code in [STIG](https://github.com/CampSmalltalk/STIG) that can be reused on another platform ... it appears to be pretty much welded into the VW infrastructure, which is should be...

It would still be very instructive for us to have a reference implementation that was ported to all of the dialects so that we could demonstrate the techniques (i.e. branch per dialect) that should be used for sharing code across dialects...

**Comment by mkobetic** [comment]
Yup bootstrapping Cypress in other dialects is something to consider. Especially bits like portable computation of differences between the files and the image can be useful. Platform specific hooks will still be required, but good chunk of it can be generic. I tried to deal with some of that in [CVST](http://sourceforge.net/projects/cvstproj), I covered that from 10k feet view in my [talk](http://www.slideshare.net/mkobetic/01-stscvst). Anyway, I'm not sure how much time I'll be able to contribute to the reference implementation, but it would be useful to have.

**Comment by dalehenrich** [comment]
My amber implementation is pretty-close to a generic implementation and relatively portable ... FileTree will be used with Pharo/Squeak/GLASS and the Amber implementation will be used with Amber and GemStone (without GLASS). I will also port the implementation to Pharo, because I need it for the Amber server part ... 

I'm hoping that someone comes along and does a GST port based on the Amber implementation.
